chimney/src/test/scala/io/scalaland/chimney/DslSpec.scala

```diff
@@ -783,6 +783,18 @@ object DslSpec extends TestSuite {
           .transformInto[shapes3.Shape] ==>
           shapes3.Rectangle(shapes3.Point(2.0, 0.0), shapes3.Point(2.0, 2.0))
       }
+
+      "fail on ambiguous targets" - {
+
+        val error = compileError(
+          """(shapes1.Triangle(shapes1.Point(0, 0), shapes1.Point(2, 2), shapes1.Point(2, 0)): shapes1.Shape)
+               .transformInto[shapes5.Shape]
+          """
+        )
+
+        error.check("", "coproduct instance Triangle of io.scalaland.chimney.examples.shapes5.Shape is ambiguous")
+        error.check("", "coproduct instance Rectangle of io.scalaland.chimney.examples.shapes5.Shape is ambiguous")
```

Comment:
That's the tricky part about which ambiguities we should report. I think this error and your implementation is correct (to report both Triangle/Rectangle) ambiguities because we're interested in finding ambiguous instances of target coproduct for all of the source coproduct instances matched by name, and limited to those. If you have any extra ambiguous instances in target (let's say we add `Polygon` occurence twice to `shapes5.Shape`), they should not be reported for such transformation. Can we somehow ensure about it in this (or separate) test?

Reply:
Added Circle occurence twice in shapes5.Shape. Updated test to check there is no error about it.